framework/src/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -587,11 +601,18 @@ public AnnotationMirror findCorrespondingAnnotation(
     public AnnotationMirror getAnnotationInHierarchy(
             Collection<? extends AnnotationMirror> annos, AnnotationMirror top) {
         AnnotationMirror annoInHierarchy = null;
-        for (AnnotationMirror rhsAnno : annos) {
-            if (isSubtype(rhsAnno, top)) {
-                annoInHierarchy = rhsAnno;
+        AnnotationMirror polyAll = null;
```

Comment:
Can both annoInHierarchy and polyAll be non-null?  (That would mean that there can be multiple annotations from the same hierarchy in the set.)